includes/classes/pmpro-testimonial-form.php

```diff
@@ -128,113 +126,160 @@ public function display( $echo = true ) {
-									$url_field = new PMPro_Field(
-										'url',
-										'text',
-										array(
-											'label'        => __( 'URL', 'pmpro-testimonials' ),
-											'required'     => in_array( 'url', $this->required_fields ),
-											'showrequired' => 'label',
-										)
-									);
-									$url_field->displayAtCheckout();
+									<?php
+									$value = '';
+									if ( ! empty( $_POST['url'] ) ) {
+										$value = wp_unslash( $_POST['url'] );
+									}
+									$required = false;
+									if ( in_array( 'url', $this->required_fields ) ) {
+										$required = true;
+									}
+									?>
+									<div class="<?php echo esc_attr( pmpro_get_element_class( 'pmpro_form_field pmpro_form_field-required pmpro_form_field-text', 'url' ) ); ?>">
+										<label for="url" class="<?php echo esc_attr( pmpro_get_element_class( 'pmpro_form_label' ) ); ?>">
+											<?php esc_html_e( 'URL', 'paid-memberships-pro' );?>
+											<?php if ( $required ) { ?><span class="<?php esc_attr_e( pmpro_get_element_class( 'pmpro_asterisk' ) ); ?>"> <abbr title="<?php esc_html_e( 'Required Field', 'pmpro-testimonials' ); ?>">*</abbr></span><?php } ?>
+										</label>
+										<input type="url" name="url" class="<?php echo esc_attr( pmpro_get_element_class( 'pmpro_form_input pmpro_form_input-text', 'url' ) ); ?>" value="<?php echo esc_attr( $value ); ?>" />
+									</div>
+
+									<?php
 
 									// Category Dropdown (if enabled).
 									if ( filter_var( $this->category_dropdown, FILTER_VALIDATE_BOOLEAN ) ) {
-										$selected_category = isset( $_POST['testimonial_category'] ) ? intval( $_POST['testimonial_category'] ) : '';
-										$category_field    = new PMPro_Field(
-											'testimonial_category',
-											'select',
+										$categories = get_terms(
 											array(
-												'label'   => __( 'Category', 'pmpro-testimonials' ),
-												'options' => wp_list_pluck(
-													get_terms(
-														array(
-															'taxonomy' => 'pmpro_testimonial_category',
-															'hide_empty' => false,
-														)
-													),
-													'name',
-													'term_id'
-												),
-												'default' => $selected_category,
+												'taxonomy' => 'pmpro_testimonial_category',
+												'hide_empty' => false,
 											)
 										);
-										$category_field->displayAtCheckout();
+										if ( $categories ) {
+											$selected_category = isset( $_POST['testimonial_category'] ) ? intval( $_POST['testimonial_category'] ) : '';
+											$required = false;
+											if ( in_array( 'url', $this->required_fields ) ) {
+												$required = true;
+											}
+											?>
+											<div class="<?php echo esc_attr( pmpro_get_element_class( 'pmpro_form_field pmpro_form_field-required pmpro_form_field-select', 'url' ) ); ?>">
+												<label for="url" class="<?php echo esc_attr( pmpro_get_element_class( 'pmpro_form_label' ) ); ?>">
```

Comment:
This should be  `for="testimonial_category"`

Reply:
Resolved this.